src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs

```diff
@@ -567,6 +567,22 @@ public NodeViewModel(WorkspaceViewModel workspaceViewModel, NodeModel logic)
             ++NoteViewModel.StaticZIndex;
         }
 
+        public virtual void Dispose()
+        {
+            //TODO add other event removal here.
+
+            this.NodeModel.PropertyChanged -= logic_PropertyChanged;
+
+            DynamoViewModel.Model.PropertyChanged -= Model_PropertyChanged;
+            DynamoViewModel.Model.DebugSettings.PropertyChanged -= DebugSettings_PropertyChanged;
+            if (IsDebugBuild)
```

Comment:
General question: Is the IsDebugBuild check necessary here? If an event is -= and the event had not been previously subscribed would this cause an issue? If not, it seems like it may possibly be safer just to unsubscribe it in case there is somewhere it is subscribed in the future outside of a IsDebugBuild check. No action necessary here, just asking a question.

Reply:
No, there should be no harm - unfortunately this call doesn't work in all cases - the engine controller gets recreated at some point when a new workspace is opened - sometimes - it is before this event is unsubscribed - that is bad.

I need to investigate this as well - thankfully this should not effect release builds.